library_management_system_downloader/run_api_download_and_parse_results.py

```diff
@@ -0,0 +1,202 @@
+
+logging.basicConfig(level=logging.INFO)
+
+# =============================================================================
+# Set up a connection to (and tables within) an SQLite database:
+# =============================================================================
+
+# Initialize the database:
+sql_engine = create_engine(
+        'sqlite:///library_coverage_xml_and_fulltext_indicators.db')
+
+
+# From the sqlalchemy documentation, at
+# http://docs.sqlalchemy.org/en/latest/dialects/sqlite.html:
+# Turn on write-ahead logging, which will allow reading and writing at
+# the same time (without this, I was getting an error re: the database being
+# locked when chunksize is used with read_sql_query, since that returns a
+# generator and is thus still reading the database:
+@event.listens_for(sql_engine, "connect")
+def set_sqlite_pragma(sql_engine, connection_record):
+    cursor = sql_engine.cursor()
+    cursor.execute("PRAGMA journal_mode=WAL")
+    cursor.close()
+
+# Create an sqlalchemy metadata schema:
+metadata = MetaData(sql_engine)
+
+# Create a schema for the table that will hold the API data:
+# The full_text_indicator column will be an integer in that 1 = fulltext
+# available and 0 = fulltext not available:
+dois_table = Table(
+        'dois_table', metadata,
+        Column('database_id', Integer, primary_key=True),
+        Column('doi', String, nullable=False, unique=True, index=True))
+
+library_holdings_table = Table(
+        'library_holdings_data', metadata,
+        Column('doi_foreign_key',
+               String, ForeignKey("dois_table.database_id"), nullable=False),
+        Column('timestamp', DateTime, nullable=False,
+               default=func.now()),  # Default is creation time
+        Column('xml_response', String, nullable=False),
+        Column('full_text_indicator', Integer))
+
+dois_table.create(
+        checkfirst=True)  # Don't create the table if it already exists.
+
+library_holdings_table.create(
+        checkfirst=True)  # Don't create the table if it already exists.
+
+# Create an sqlalchemy portal for inserting new records into
+# the tables:
+dois_table_insert = dois_table.insert()
+holdings_table_insert = library_holdings_table.insert()
+
+# =============================================================================
+# Define functions for querying and inserting into the database
+# =============================================================================
+
+Session = sessionmaker(bind=sql_engine)
+sql_session = Session()
+
+
+def is_doi_already_answered_in_database(
+        doi,
+        sqlalchemy_session=sql_session):
+    existing_doi_with_fulltext_anwer = sql_session.query(distinct(
+        dois_table.c.doi)).join(library_holdings_table).filter(
+            dois_table.c.doi == doi).filter(
+                    library_holdings_table.c.full_text_indicator.isnot(
+                            None)).first()
```

Comment:
Fixed the spelling error and changed the indentation in aad46c7a.

Reply:
...and fixed that commit in 7d6ade0, where I also added a new config. setting, `oadoi_values_to_limit_search_to`, to streamline the process of choosing which DOIs to download (this was previously hardcoded in the script itself).

Reply:
..Annnd fixed a typo in 2b00485.
I found that Spyder/iPython was erroring out with of an indentation error until I added the slashes at the end of each line here, so that's why I added them.